### PR TITLE
Get com coords

### PR DESCRIPTION
**Pull request by ArjunPutcha**
Create projection files in for loop using getattr()
extract CoM coords while also getting contour coords

### CODE REVIEW

File: mesmerize_core/algorithms/cnmf.py [hunk 1]
```diff
@@ -58,12 +58,13 @@ def main(batch_path, uuid, data_path: str = None):
         Yr, dims, T = cm.load_memmap(fname_new)
         images = np.reshape(Yr.T, [T] + list(dims), order='F')
 
-        mean_projection_path = str(Path(input_movie_path).parent.joinpath(f'{uuid}_mean_projection.npy'))
-        std_projection_path = str(Path(input_movie_path).parent.joinpath(f'{uuid}_std_projection.npy'))
-        max_projection_path = str(Path(input_movie_path).parent.joinpath(f'{uuid}_max_projection.npy'))
-        np.save(mean_projection_path, np.mean(images, axis=0))
-        np.save(std_projection_path, np.std(images, axis=0))
-        np.save(max_projection_path, np.max(images, axis=0))
+        paths = []
```
Comment (kushalkolar) [inline mesmerize_core/algorithms/cnmf.py:61]: use a dict for this

File: mesmerize_core/algorithms/cnmf.py [hunk 2]
```diff
@@ -114,9 +115,9 @@ def main(batch_path, uuid, data_path: str = None):
                 "cnmf-hdf5-path": cnmf_hdf5_path,
                 "cnmf-memmap-path": cnmf_memmap_path,
                 "corr-img-path": corr_img_path,
-                "mean-projection-path": mean_projection_path,
-                "std-projection-path": std_projection_path,
-                "max-projection-path": max_projection_path,
+                "mean-projection-path": paths[0],
```
Comment (kushalkolar) [inline mesmerize_core/algorithms/cnmf.py:118]: use a dict for this

File: mesmerize_core/caiman_extensions/cnmf.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 from caiman.source_extraction.cnmf import CNMF
 from caiman.source_extraction.cnmf.cnmf import load_CNMF
 from caiman.utils.visualization import get_contours as caiman_get_contours
+from caiman.base.rois import com
```
Comment (kushalkolar) [inline mesmerize_core/caiman_extensions/cnmf.py:11]: you don't need this

File: mesmerize_core/caiman_extensions/cnmf.py [hunk 4]
```diff
@@ -135,7 +136,9 @@ def _get_spatial_contour_coors(cnmf_obj: CNMF):
             swap_dim=True
         )
 
-        return contours
+        COM = com(cnmf_obj.estimates.A, d1=dims[1], d2=dims[0])
```
Comment (kushalkolar) [inline mesmerize_core/caiman_extensions/cnmf.py:139]: remove this

File: mesmerize_core/caiman_extensions/cnmf.py [hunk 5]
```diff
@@ -135,7 +136,9 @@ def _get_spatial_contour_coors(cnmf_obj: CNMF):
...
+
+        return contours, COM
```
Comment (kushalkolar) [inline mesmerize_core/caiman_extensions/cnmf.py:141]: remove COM

File: mesmerize_core/caiman_extensions/cnmf.py [hunk 6]
```diff
@@ -152,25 +155,25 @@ def get_spatial_contours(self, ixs_components: np.ndarray) -> List[dict]:
 
         """
         cnmf_obj = self.get_output()
-        contours = self._get_spatial_contour_coors(cnmf_obj)
+        contours, COM = self._get_spatial_contour_coors(cnmf_obj)
```
Comment (kushalkolar) [inline mesmerize_core/caiman_extensions/cnmf.py:158]: remove COM

File: mesmerize_core/caiman_extensions/cnmf.py [hunk 7]
```diff
@@ -152,25 +155,25 @@ def get_spatial_contours(self, ixs_components: np.ndarray) -> List[dict]:
...
 
         contours_selection = list()
         for i in range(len(contours)):
             if i in ixs_components:
                 contours_selection.append(contours[i])
 
-        return contours_selection
+        return contours_selection, COM
```
Comment (kushalkolar) [inline mesmerize_core/caiman_extensions/cnmf.py:165]: remove COM

File: mesmerize_core/caiman_extensions/cnmf.py [hunk 8]
```diff
@@ -152,25 +155,25 @@ def get_spatial_contours(self, ixs_components: np.ndarray) -> List[dict]:
...
-
+        contours, centres_of_mass = self.get_spatial_contours(ixs_components)
         coordinates = []
         for contour in contours:
             coors = contour['coordinates']
             coordinates.append(coors[~np.isnan(coors).any(axis=1)])
 
-        return coordinates
+
+        return coordinates, centres_of_mass
```
Comment (kushalkolar) [inline mesmerize_core/caiman_extensions/cnmf.py:176]:
```python
return coordinates, contour['CoM']
```

**Review by kushalkolar — COMMENTED** [review]
minor changes

File: mesmerize_core/caiman_extensions/cnmf.py (hunk 4 above)
Comment (YichengZou626) [reply mesmerize_core/caiman_extensions/cnmf.py:139]: this does not work, and need to modify caiman_get_contour

**Pull request merged**